File: integration/4_grpc/random_server.py
from concurrent import futures
import logging
import random

import grpc
import random_pb2
import random_pb2_grpc


class RandomServer( random_pb2_grpc.RandomServiceServicer ):

    def SetBounds( self, request, context ):
        # TODO
        return

    def NextRandom( self, request, context ):
        # TODO
        return

def serve():
    server = grpc.server( futures.ThreadPoolExecutor( max_workers=10 ))
    random_pb2_grpc.add_RandomServiceServicer_to_server( RandomServer(), server )
    server.add_insecure_port( '[::]:50051' )
    server.start()
    print( "server ready" )
    server.wait_for_termination()


if __name__ == '__main__':
    logging.basicConfig()
    serve()

File: integration/4_grpc/random_server.cc

#include <iostream>
#include <string>
#include <memory>

#include <cstdlib>

#include <grpc++/grpc++.h>

#include "random.pb.h"
#include "random.grpc.pb.h"

using grpc::Server;
using grpc::ServerBuilder;
using grpc::ServerContext;
using grpc::Status;

namespace if3040::grpc {

class RandomServer final : public RandomService::Service {
public:
    RandomServer() {}
    
    Status SetBounds( ServerContext * context,
                      const SetBoundsRequest *  request,
                      SetBoundsResponse  * response ) {
        // TODO
        return Status::OK;
    }
    
    Status NextRandom( ServerContext * context,
                       const NextRandomRequest * request,
                       NextRandomResponse * response ) {
        // TODO
        return Status::OK;
    }

private:
    // TODO
};

}


int main() {
    if3040::grpc::RandomServer random_server;
    
    std::string server_address( "0.0.0.0:50051" );
    ServerBuilder builder;
    builder.AddListeningPort( server_address, grpc::InsecureServerCredentials() );
    builder.RegisterService( &random_server );
    std::unique_ptr< Server > server( builder.BuildAndStart() );
    std::cout << "Server listening on " << server_address << std::endl;
    server->Wait();
    
    return 0;
}
